### PR TITLE
Introduce autoscalers for inbox listener and stream service

### PR DESCRIPTION
**Pull request by wkalt**
Adds autoscalers for the inbox-listener and stream service. The inbox
listener autoscaler relies on KEDA. The stream service autoscaler is a
standard HPA.

KEDA has some unfortunate friction with helm. It cannot be added as a
dependency to the chart and installed in a reasonable way, because it
relies on CRDs, which helm will not be responsible for updating or
installing in order.

Based on conversation in https://github.com/kedacore/charts/issues/226,
there are two alternative approaches open to us:

* We can instruct users to install KEDA on their cluster prior to
  installing our deployment. The command for this is,
    helm install keda --version 2.9.1 --namespace keda kedacore/keda --create-namespace

* We can vendor KEDA's CRDs and put them into our chart, under a crd
  directory. This should enable a single-command install, but comes with
  the downsides that,
  1. Helm will not update or manage the keda CRDs for us
  2. If users already have KEDA installed (certainly possible) we will
     encounter a conflict.

  Mainly due to the second item, this PR takes the first approach.

### CODE REVIEW

**Comment by wkalt** [comment]
I think we can go further with the environment overrides - there is no need for our values file to define the azure-specific stuff like storageAccountName, serviceUrl. These are just details on how the user wants to auth their pods to a cloud provider, rather than part of our chart.

We can remove those from values.yaml and instead instruct users to supply env vars as necessary. Similarly, I don't think "gcp-cloud-credential" needs to appear in our charts, we can instead allow the user to mount volumes via templates, using their values file. I have not changed this in the current patch though.

File: charts/primary-site/templates/garbage-collector-cronjob.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 apiVersion: batch/v1
 kind: CronJob
 metadata:
-  name: garbage-collector
+  name: {{ .Values.garbageCollector.name }}
```
Comment (defunctzombie) [inline charts/primary-site/templates/garbage-collector-cronjob.yaml:4]: Why do we need to template this?

File: charts/primary-site/templates/garbage-collector-cronjob.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 apiVersion: batch/v1
```
Comment (defunctzombie) [inline charts/primary-site/templates/garbage-collector-cronjob.yaml:1]: Why are all these moved out of the folders? Is there some problem with the folders?

File: charts/primary-site/values.yaml [hunk 3]
```diff
@@ -11,33 +10,48 @@ globals:
...
         cpu: 1000m
         memory: 1Gi
       limits:
         cpu: 1000m
         memory: 1Gi
+  env:
 
 garbageCollector:
+  name: garbage-collector
   schedule: "*/10 * * * *" # every 10 minutes
```
Comment (defunctzombie) [inline charts/primary-site/values.yaml:54]: should these three items (schedule, failedJobs, successfulJobs) be under a `deployment` section?

**Review by defunctzombie — APPROVED** [review]
Not sure why we did away with the folders - those seemed nice? Do other helm charts in the wild use folders?

If you want to make names configurable I think you'll need to update the `stream-service-svc.yaml` file to reference a name as well.

What's the purpose of the `env` override? Is this a common pattern? I think we should remove this and expose configuration options explicitly.

Does this fail gracefully if keda is not installed? Do we need to allow the user to select a namespace other than `keda` ?

**Comment by defunctzombie** [comment]
We're going in a different direction for now: https://github.com/foxglove/helm-charts/pull/28

**Pull request closed without merging**